python/lsst/afw/detection/utils.py

```diff
@@ -49,3 +54,51 @@ def writeFootprintAsDefects(fd, foot):
     y0:     %4d                         # Starting row
     height: %4d                         # number of rows
 }""" % (bbox.getMinX(), bbox.getWidth(), bbox.getMinY(), bbox.getHeight()), file=fd)
+
+
+def footprintsToNumpy(
+    catalog: SourceCatalog,
+    bbox: geom.Box2I | None = None,
+    shape: tuple[int, int] | None = None,
+    xy0: tuple[int, int] | None = None,
+    asBool: bool = True,
+) -> np.ndarray:
+    """Convert all of the footprints in a catalog into a boolean array.
+
+    Parameters
+    ----------
+    catalog:
+        The source catalog containing the footprints.
+        This is typically a mergeDet catalog, or a full source catalog
+        with the parents removed.
+    shape:
+        The final shape of the output array.
+    xy0:
+        The lower-left corner of the array that will contain the spans.
+
+    Returns
+    -------
+    result:
+        The array with pixels contained in `spans` marked as `True`.
+    """
+    if bbox is None and shape is None:
+        raise RuntimeError("Must provide either bbox or shape")
+
+    if bbox is not None:
+        width, height = bbox.getDimensions()
+        shape = (height, width)
+        xy0 = (bbox.getMinX(), bbox.getMinY())
+
+    if xy0 is None:
+        offset = (0, 0)
+    else:
+        offset = (-xy0[0], -xy0[1])
+
+    result = np.zeros(shape, dtype=bool)
```

Comment:
`result = np.zeros(shape, dtype=bool if asBool else int)` (or else I don't think this can work correctly if `asBool = False`)